### ISSUE TITLE
Excluding files from the repository

### ISSUE THREAD

**Issue by SylvainCorlay** [jupyter/repo2docker#937]
Is there a means to exclude files of the git repository from the docker image?

The use case is that of a (private) repository including solution notebooks to an assignment.

I presume this could be achieved with a post-build or so, but simply asking if we could have a configuration thing to include e.g. directories from the build (as well as the .git folder I guess).

**Comment by dolfinus** [comment]
You can add `.dockerignore` file into the root of your repo. This file is used by docker for the same purpose as `.gitignore` file by git.

**Comment by yuvipanda** [comment]
@dolfinus I don't think repo2docker respects .dockerignore files.

**Comment by dolfinus** [comment]
That's unfortunate. Without respecting .dockerignore repo2docker add all the from the repo folder files into the docker image, including .git folder. This is messing up with the docker layer caching mechanism and also increasing the resulting image on every pushed commit

**Comment by yuvipanda** [comment]
Yeah, I think adding .dockerignore support would be a good addition to repo2docker functionality. I'll retitle the issue.
